statbot/sql.py

```diff
@@ -285,11 +318,28 @@ def _user_values(user, deleted=False):
             'user_id': user.id,
             'name': user.name,
             'discriminator': user.discriminator,
+            'avatar': user.avatar,
             'is_deleted': deleted,
             'is_bot': user.bot,
         }
 
     @staticmethod
+    def _nick_values(member):
+        return {
+            'user_id': member.id,
+            'guild_id': member.guild.id,
+            'nickname': member.nick,
+        }
+
+    @staticmethod
+    def _role_member_values(member, role):
+        return {
+            'role_id': role.id,
+            'guild_id': role.guild.id,
+            'user_id': member.id,
+        }
```

Comment:
why do these need to be their own functions? (especially if we got rid of `add_member`)

Reply:
I did it because then all the object -> SQL mapping occurs in one place (per object). This way if certain extracted attributes are needed in multiple places, I don't have duplicated code that would require extra changes if the schema or API objects are ever changed.

Reply:
oh god I didn't realize there were waaaaaaaaaaaay more `_values` functions